rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Fixture/OneToMany/Attribute/skip_collection_intersection_selectable.php.inc

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Rector\Doctrine\Tests\CodeQuality\Rector\Property\ImproveDoctrineCollectionDocTypeInEntityRector\Fixture\OneToMany\Attribute;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Selectable;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Rector\Doctrine\Tests\CodeQuality\Rector\Property\ImproveDoctrineCollectionDocTypeInEntityRector\Source\CountryRef;
+
+#[ORM\Entity]
+class SkipCollectionIntersectionSelectable
+{
+	/**
+	 * @var Collection<int, CountryRef>&Selectable
+	 */
+	#[ORM\OneToMany(mappedBy: 'country', targetEntity: CountryRef::class, cascade: ['persist'])]
+	private Collection $countryRefs;
```

Comment:
While this fixes proposed issue, it doesn't make sense from PHP points of view.

This code should be updated as well:

```php
private Collection $countryRefs;
```

Also, in the case of using `matching()` in many places, this would be correct:

```php
private ArrayCollection $countryRefs;
```

But this is out of scope of this rule. I'd recommend to create custom rule to use `ArrayCollection` if possible.

Reply:
It is not possible to use `ArrayCollection` as type.  Depending on the state of the object, it is either `ArrayCollection` or `PersistentCollection`.